src/App.tsx

```diff
@@ -1,11 +1,19 @@
-import { useState } from 'react';
+import { useCallback, useState } from 'react';
 import './App.scss';
 import { MoviesList } from './components/MoviesList';
 import { FindMovie } from './components/FindMovie';
 import { Movie } from './types/Movie';
 
 export const App = () => {
-  const [movies] = useState<Movie[]>([]);
+  const [movies, setMovies] = useState<Movie[]>([]);
+
+  const addMovie = useCallback((movie: Movie) => {
+    const hasMovie = movies.some(m => m.imdbId === movie.imdbId);
+
+    if (!hasMovie) {
```

Comment:
it would be nice to show message, why we won't add movie into list, in case of it's already there